core/modules/field_ui/field_ui.admin.inc

```diff
@@ -1514,7 +1514,9 @@ function field_ui_widget_type_options($field_type = NULL, $by_label = FALSE) {
           $options[$widget_field_type][$name] = $widget_type['label'];
         }
       }
-      asort($options[$widget_field_type]);
+      if (!empty($options)) {
```

Comment:
Should this be checking `$options[$widget_field_type]`? I could see `$options` being set for some but not all of the widget types in the `foreach()` loop we're in side of.

```suggestion
      if (isset($options[$widget_field_type])) {
```

Reply:
Doesn't that duplicate the check from line 1513? The loop above only adds the item to the `$options`array 
`if (isset($field_types[$widget_field_type])). So at that point (line 1517) I'd assume that any entry in the array has both, name and label set properly. Or am I missing something?

Reply:
@quicksketch any chance you find the time to look at my answer comment?

Reply:
>  The loop above only adds the item to the `$options` array `if (isset($field_types[$widget_field_type]))`

Other values might already be in `$options` from a previous loop. There are two levels of looping here. It's possible that the first pass on `foreach (field_info_widget_types() as $name => $widget_type) {` added something into `$options` already, and the second pass may not have put something in `$options[$widget_field_type]`. Even if that might not occur in common scenarios, it seems like we'd be safer checking the exact key exists before attempting to sort its value.